.travis.yml

```diff
@@ -0,0 +1,19 @@
+services:
+- docker
+script:
+# Ideally we'd just run compile.sh here:
+#
+# - VERBOSE=yes bash ./compile.sh
```

Comment:
I think here is the big misunderstanding of that pull request. Simply calling `bash compile.sh` is not a supported way of building bazel. So, as it is not supposed to work, there is nothing to fix here. The [supported ways of building bazel from source]( https://docs.bazel.build/versions/master/install-compile-source.html) are
- compiling from the distribution artifact, which on top of the sources also has the required generated `protoc` output. (That is used to build a bootstrap bazel that then builds protoc and bazel from sources). This the quickest and easiest way to bootstrap bazel on a new system
- have `protoc` already installed on your system and telling `compile.sh` about its location. This is the way to cooperate with an ambient system providing the needed tools, to leave the way open to eventually build bazel without depending on any third-party tools embedded in our source tree
- have an older version of `bazel` already and `bazel build //src:bazel`; this is the typical workflow for `bazel` developers.

The the [design document](https://www.bazel.build/designs/2016/10/11/distribution-artifact.html) for a background discussion.

Side remark: bazel is currently not using travis.

Reply:
I actually overlooked the ddoc, thanks for the link!

My proposal is to remove any binaries from the repo / releases (which addresses all of the ddoc's identified shortcomings), and supporting only 2 or 3 ways of building *from sources*:
- with an existing old bazel
- with an existing "correct version" of protoc + grpc-java/compiler
- with without any of these (but with a "standard" opensource / unix build toolchain)

(and incidentally to fix the current scenario 2 - relying on the distro w/ prebuilt protoc - if it's indeed broken as I think it is)